dlt/cli/deploy_command.py

```diff
@@ -33,188 +34,367 @@
 REQUIREMENTS_GITHUB_ACTION = "requirements_github_action.txt"
 GITHUB_URL = "https://github.com/"
 DLT_DEPLOY_DOCS_URL = "https://dlthub.com/docs/walkthroughs/deploy-a-pipeline"
+DLT_AIRFLOW_GCP_DOCS_URL = "https://dlthub.com/docs/running-in-production/orchestrators/airflow-gcp-cloud-composer"
+AIRFLOW_GETTING_STARTED = "https://airflow.apache.org/docs/apache-airflow/stable/start.html"
+AIRFLOW_DAG_TEMPLATE_SCRIPT = "dag_template.py"
+AIRFLOW_CLOUDBUILD_YAML = "cloudbuild.yaml"
 
 
-def deploy_command(pipeline_script_path: str, deployment_method: str, schedule: str, run_on_push: bool, run_on_dispatch: bool, branch: str = None) -> None:
+class DeploymentMethods(Enum):
+    github_actions = "github-action"
+    airflow = "airflow"
+
+
+def deploy_command(
+    pipeline_script_path: str,
+    deployment_method: str,
+    schedule: Optional[str],
+    run_on_push: bool,
+    run_on_dispatch: bool,
+    branch: Optional[str] = None
+) -> None:
     # get current repo local folder
     with get_repo(pipeline_script_path) as repo:
-        repo_storage = FileStorage(str(repo.working_dir))
+        deployment_state = DeploymentState(repo, pipeline_script_path, schedule, run_on_push, run_on_dispatch, branch)
+        deployment_state.run_deploy(deployment_method)
+
+
+class DeploymentState:
+    def __init__(
+        self,
+        repo,
+        pipeline_script_path: str,
+        schedule: Optional[str],
+        run_on_push: bool,
+        run_on_dispatch: bool,
+        branch: Optional[str] = None
+    ):
+        self.repo = repo
+        self.pipeline_script_path = pipeline_script_path
+        self.schedule = schedule
+        self.run_on_push = run_on_push
+        self.run_on_dispatch = run_on_dispatch
+        self.branch = branch
+
+        self.repo_storage = FileStorage(str(self.repo.working_dir))
```

Comment:
it's better to move actual code to some kind of `_prepate_deployment` method that will be called by `run_deployment`. `__init__` should not do too much logic